onnxruntime/core/session/inference_session.cc

```diff
@@ -1505,6 +1505,134 @@ common::Status InferenceSession::ValidateOutputs(const std::vector<std::string>&
   return common::Status::OK();
 }
 
+#ifdef ENABLE_TRAINING
+std::pair<size_t, size_t> InferenceSession::GetBreakpointAndEndPoint() {
+  size_t end_point;
+  size_t break_point = 0;
+  const SequentialExecutionPlan& seq_exec_plan = *(session_state_->GetExecutionPlan());
+  const auto& exec_plan_vec = seq_exec_plan.execution_plan;
+  end_point = exec_plan_vec.size() - 1;
+  for (size_t program_counter = 0; program_counter < exec_plan_vec.size(); program_counter += 1) {
+    const auto& node_exec_plan = exec_plan_vec[program_counter];
+    auto node_index = node_exec_plan.node_index;
+    if (session_state_->GetKernel(node_index)->KernelDef().OpName() == "YieldOp") {
+      break;
+    }
+    break_point += 1;
+  }
+  return std::make_pair(break_point, end_point);
+}
+
+Status InferenceSession::Run(onnxruntime::RunOptions& run_options, std::vector<OrtValue>& feeds, std::vector<OrtValue>& fetches,
```

Comment:
Can you rename this method to `PartialRun()`? Should make it easier to distinguish from the existing overloads when reading the code.